core/src/lib.rs

```diff
@@ -62,13 +62,15 @@
 //! }
 //! ```
 
+pub mod get_similar_mail_provider;
```

Comment:
> In contrast to your suggestion, I moved the check inside the lib-function to avoid some duplication. Let me know what you think :)

Could you expand on which duplication you're referring to? I still think this should be inside syntax

Reply:
Ah, in the issue, it seemed like you'd mean outside the whole check. If we'd do it inside the syntax check, we could also catch the blacklisted ones you mentioned below. But we wouldn't have access to the MX and SMTP results, of course